credentials/endpointcreds/internal/client/client_test.go

```diff
@@ -0,0 +1,197 @@
+				}
+			},
+			ExpectResult: &GetCredentialsOutput{
+				AccessKeyID:     "FooKey",
+				SecretAccessKey: "FooSecret",
+			},
+		},
+		"json error response": {
+			ResponseCode: 403,
+			ResponseBody: []byte(`{
+  "code": "Unauthorized",
+  "message": "not authorized for endpoint"
+}`),
+			ExpectErr: true,
+			ValidateError: func(t *testing.T, err error) (ok bool) {
+				t.Helper()
+				var apiError smithy.APIError
+				if errors.As(err, &apiError) {
+					if e, a := "Unauthorized", apiError.ErrorCode(); e != a {
+						t.Errorf("expect %v, got %v", e, a)
+						ok = false
+					}
+					if e, a := "not authorized for endpoint", apiError.ErrorMessage(); e != a {
+						t.Errorf("expect %v, got %v", e, a)
+						ok = false
+					}
+				} else {
+					t.Errorf("expect %T error type, got %T: %v", apiError, err, err)
+					ok = false
+				}
+				return ok
+			},
+		},
+		"non-json error response": {
+			ResponseCode: 500,
+			ResponseBody: []byte(`<html><body>unexpected message format</body></html>`),
+			ExpectErr:    true,
+			ValidateError: func(t *testing.T, err error) (ok bool) {
+				t.Helper()
+				if e, a := "failed to decode error message", err.Error(); !strings.Contains(a, e) {
+					t.Errorf("expect %v, got %v", e, a)
+					ok = false
+				}
+				return ok
+			},
+		},
+	}
+
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+			var actualReq *http.Request
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				actualReq = r.Clone(r.Context())
+				var buf bytes.Buffer
+				if _, err := io.Copy(&buf, r.Body); err != nil {
+					t.Errorf("failed to read r body, %v", err)
+				}
+
+				actualReq.Body = ioutil.NopCloser(bytes.NewReader(buf.Bytes()))
+
+				w.WriteHeader(tt.ResponseCode)
+				w.Write(tt.ResponseBody)
+			}))
+			defer server.Close()
+
+			client := New(Options{
+				AuthorizationToken: tt.Token,
+				Endpoint:           server.URL + tt.RelativeURI,
+			})
+
+			result, err := client.GetCredentials(context.Background())
+			if (err != nil) != tt.ExpectErr {
+				t.Fatalf("got error %v, but ExpectErr=%v", err, tt.ExpectErr)
+			}
+			if err != nil && tt.ValidateError != nil {
+				if !tt.ValidateError(t, err) {
+					return
+				}
+			}
+
+			if e, a := "application/json", actualReq.Header.Get("Accept"); e != a {
+				t.Errorf("expect %v, got %v", e, a)
+			}
+
+			if len(tt.Token) > 0 {
+				if e, a := tt.Token, actualReq.Header.Get("Authorization"); e != a {
+					t.Errorf("expect %v, got %v", e, a)
+				}
+			}
+
+			if tt.ValidateRequest != nil {
+				tt.ValidateRequest(t, actualReq)
+			}
+
+			if err = smithytesting.CompareValues(tt.ExpectResult, result); err != nil {
+				t.Errorf("expect value match:\n%v", err)
+			}
+		})
+	}
+}
+
+func TestClient_GetCredentials_NoEndpoint(t *testing.T) {
+	client := New(Options{})
+
+	_, err := client.GetCredentials(context.Background())
+	if err == nil {
+		t.Fatalf("expect ~ got none")
+	}
```

Comment:
is the `~` here intentional?  Guessing might be hold over from template

Reply:
I think this was accidental 👍